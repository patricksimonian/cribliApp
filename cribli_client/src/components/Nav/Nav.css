#sidebar-main {
  position: fixed;
  top: 0;
  left: 0;
  width: 55px;
  height: 100%;
  z-index: 5;
  background-color: #3e4963;
  color: #fff;
  font-size: 16px;
}

.sidebar-section {
  padding: 5px;
}

.sidebar-upper {
  height: 60%;
}

.sidebar-lower {
  width: 60%;
}

.nav-icon-action {
  font-size: 2em;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  transition: color .15s ease-in-out, transform .10s ease-in-out;
  background: none;
  border: none;

}
.nav-icon-action:not(:first-child) {
  margin: 10px;
}
.nav-icon-action:focus,
.nav-icon-action:active,
.nav-icon-action:visited {
  color: inherit;
  outline: none;
}
.nav-icon-action:hover {
  color: #b1e1ff;
  transform: translateX(2px);
}

.action-text {
  font-size: .4em;
  letter-spacing: 1px;
  margin-top: 5px;
}
.nav-icon-wrap {
  position: relative;
}
.nav-icon-wrap.menu-close .nav-icon-menu {
  display: none;
}
.nav-icon-wrap.menu-open .nav-icon-menu {
  display: flex;
}

.nav-icon-menu {
  position: fixed;
  left: 55px;
  top: 0;
  background: #1B5299;
}

