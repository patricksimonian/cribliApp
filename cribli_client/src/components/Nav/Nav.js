// ItemService.js
import React, { Component } from "react";
import "./Nav.css"
// import NavIcon from "../NavUtils/NavIcon/NavIcon";
import Login from "../Login/Login.js";
class Nav extends Component {
 constructor(props) {
  super(props);

 }

 _handleHover(menu, currentState) {
  let className = currentState === 'close' ? 'open' : 'close';
  console.log(menu, currentState)
  this.props.changeMenuClass(menu, className);
 }
 render() {
  return (
      <nav id="sidebar-main">
        <div className="sidebar-upper sidebar-section">
          <div className={"nav-icon-wrap menu-" + this.props.menus[0][1]}>
            <button className="nav-icon-action" onClick={(e) => {
              e.preventDefault();
              this._handleHover("login", this.props.menus[0][1])
            }}>
              <i className="fa fa-sign-in" aria-hidden="true" />
              <span className="action-text">sign-in</span>
            </button>
            <div className="nav-icon-menu">
              <Login />
            </div>
          </div>
        </div>
        <div className="sidebar-lower sidebar-section">

        </div>
      </nav>
    );
 }
}

export default Nav;
