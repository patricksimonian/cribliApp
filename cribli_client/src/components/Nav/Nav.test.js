import React from 'react';
import Nav from './Nav';
import renderer from 'react-test-renderer';


it('renders with the right amount of icons', () => {
  const component = renderer.create(
      <Nav />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
