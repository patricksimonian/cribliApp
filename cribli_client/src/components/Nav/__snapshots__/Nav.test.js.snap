// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the right amount of icons 1`] = `
<nav
  id="sidebar-main"
>
  <div
    className="sidebar-upper sidebar-section"
  >
    <div
      className="nav-icon-wrap"
    >
      <a
        className="nav-icon-action"
        href=""
      >
        <i
          aria-hidden="true"
          className="fa fa-sign-in"
        />
        <span
          className="action-text"
        >
          sign-in
        </span>
      </a>
    </div>
    <div
      className="nav-icon-wrap"
    >
      <a
        className="nav-icon-action"
        href=""
      >
        <i
          aria-hidden="true"
          className="fa fa-rocket"
        />
        <span
          className="action-text"
        >
          monies
        </span>
      </a>
    </div>
  </div>
  <div
    className="sidebar-lower sidebar-section"
  />
</nav>
`;
